### ISSUE TITLE
urxvtd disco

### ISSUE THREAD

**Issue by jeroentbt** [felixr/urxvt-theme-switch#4]
I've been using the color-themes extension succesfully with urxvt. 
I love it. Thank you!

When switching over to urxvtd/c I notice this weird behaviour: http://i.imgur.com/xGuo65T.gif
1 in 2 incantations of urxvtc result in a terminal where color-themes can not find the themes.

Here's the relevant part of my `~/.Xresources`:

```
URxvt.perl-ext-common: default,matcher,keyboard-select,url-select,color-themes,clipboard
URxvt.keysym.M-C-n:  perl:color-themes:next
URxvt.keysym.M-C-p:  perl:color-themes:prev
URxvt.color-themes.themedir : /home/jeroen/.themes/urxvt
URxvt.color-themes.state-file: /home/jeroen/.urxvt-theme
URxvt.color-themes.autosave: 1
URxvt.keysym.M-C-l:  perl:color-themes:load-state
URxvt.keysym.M-C-s:  perl:color-themes:save-state
```

When doing M-C-n in a white terminal I get:
`No themes found in`

Any ideas?
Anybody else having this issue?

**Comment by willjp** [comment]
yes, I'm having this issue as well. Specifically, every third terminal I open, the script seems unable to find the themes. Afterwards then it's fine for another 3 terminals. I wish I was a little more versed in perl. Maybe it's time I take a deeper look into Perl :). I'll post if I can figure something out.

**Comment by felixr** [comment]
I can reproduce this issue, but I don't see an easy fix for this. It seems to be a problem with on_start and how it is called when you use urxvtd/urxvtc. The docs aren't very clear about it; I could not find anything helpful. Maybe it is even a bug in urxvt?

A possible solution for the disco effect is to hardcode the save file instead of using the settings. The 'No themes found' problem could be fixed in a similar way: hard coding the theme dir and reading the list of themes in on_user_command.
